src/main/java/org/checkerframework/checker/dividebyzero/DivByZeroTransfer.java

```diff
@@ -49,51 +60,196 @@ enum BinaryOperator {
      * input value "lhs" is always a legal return value, but not a very useful
      * one.)
      *
-     * <p>For example, given the code
+     * <p>
+     * For example, given the code
+     * 
      * <pre>
-     * if (y != 0) { x = 1 / y; }
+     * if (y != 0) {
+     *     x = 1 / y;
+     * }
      * </pre>
+     * 
      * the comparison "y != 0" causes us to learn the fact that "y is not zero"
      * inside the body of the if-statement. This function would be called with
      * "NE", "top", and "zero", and should return "not zero" (or the appropriate
      * result for your lattice).
      *
-     * <p>Note that the returned value should always be lower in the lattice
+     * <p>
+     * Note that the returned value should always be lower in the lattice
      * than the given point for lhs. The "glb" helper function below will
      * probably be useful here.
      *
-     * @param operator   a comparison operator
-     * @param lhs        the lattice point for the left-hand side of the comparison expression
-     * @param rhs        the lattice point for the right-hand side of the comparison expression
+     * @param operator a comparison operator
+     * @param lhs      the lattice point for the left-hand side of the comparison
+     *                 expression
+     * @param rhs      the lattice point for the right-hand side of the comparison
+     *                 expression
      * @return a refined type for lhs
      */
     private AnnotationMirror refineLhsOfComparison(
             Comparison operator,
             AnnotationMirror lhs,
             AnnotationMirror rhs) {
-        // TODO
-        return lhs;
+        AnnotationMirror zero = reflect(Zero.class);
+        AnnotationMirror nonZero = reflect(NonZero.class);
+        switch (operator) {
+            case EQ:
+                if (equal(lhs, top())) {
+                    if (equal(rhs, zero)) {
+                        return zero;
+                    } else if (equal(rhs, nonZero)) {
+                        return nonZero;
+                    }
+                } else if (equal(rhs, top())) {
+                    if (equal(lhs, zero)) {
+                        return zero;
+                    } else if (equal(lhs, nonZero)) {
+                        return nonZero;
+                    }
+                }
```

Comment:
because this doesn't return unconditionally, these `case` blocks should have `break`s at the end